x-pack/test/detection_engine_api_integration/security_and_spaces/tests/create_index.ts

```diff
@@ -66,10 +65,15 @@ export default ({ getService }: FtrProviderContext) => {
 
     describe('t1_analyst', () => {
       const role = ROLES.t1_analyst;
+
       beforeEach(async () => {
         await createUserAndRole(security, role);
       });
 
+      afterEach(async () => {
```

Comment:
These have got me thinking generally about how we can guarantee these kinds of hooks to be symmetric. My first thought is helper functions that generate these hooks, e.g. 

```js
const withUserRole = (role, security) => { beforeEach(); afterEach(); }
```
 but maybe that's too much abstraction? Just a thought.

Reply:
I think this is a neat idea 👍

Reply:
Not tracking all the way on this? How do you use those functions like you have above. I can't call beforeEach and then afterEach like that. I might be dense here though.